src/main/java/io/jenkins/plugins/analysis/core/views/SourceDetail.java

```diff
@@ -66,109 +61,161 @@ public SourceDetail(final Run<?, ?> owner, final Reader affectedFile, final Issu
+            case "scala": case "sc":
+                return "language-scala";
+        }
 
-        return writer.toString();
+        return "language-clike"; //Best effort for unknown extensions
     }
 
     /**
-     * Splits the source code into three blocks: the line to highlight and the source code before and after this line.
+     * This function build
      *
-     * @param sourceFile
-     *         the source code of the whole file as rendered HTML string
+     * @param sourceFileReader Source file containing the source code to be displayed.
+     * @param fileName The filename associated to the content. It will be use to select the syntax highlight.
+     * @return
+     * @throws IOException
      */
-    // CHECKSTYLE:CONSTANTS-OFF
-    @SuppressWarnings("PMD.ConsecutiveLiteralAppends")
-    private String splitSourceFile(final String sourceFile) {
-        StringBuilder output = new StringBuilder(sourceFile.length());
+    private String buildCodeBlock(Reader sourceFileReader, String fileName) throws IOException {
 
-        LineIterator lineIterator = IOUtils.lineIterator(new StringReader(sourceFile));
+        StringWriter writer = new StringWriter();
 
-        // TODO: add support for line ranges (or replace this view with client side rendering)
-        try {
-            int lineNumber = 1;
-            while (lineNumber < SOURCE_GENERATOR_OFFSET) {
-                copyLine(output, lineIterator);
-                lineNumber++;
-            }
-            lineNumber = 1;
-            boolean isFirstRange = true;
-            while (lineNumber < issue.getLineStart()) {
-                copyLine(output, lineIterator);
-                lineNumber++;
-            }
-            output.append("</code>\n");
-            output.append("</td></tr>\n");
-            output.append("<tr><td style=\"background-color:");
-            appendRangeColor(output, isFirstRange);
-            output.append("\">\n");
-            output.append("<div id=\"line");
-            output.append(issue.getLineStart());
-            output.append("\" tooltip=\"");
-            if (issue.getLineStart() > 0) {
-                outputEscaped(output, issue.getMessage());
-            }
-            outputEscaped(output, description);
-            output.append("\" nodismiss=\"\">\n");
-            output.append("<code><b>\n");
-            if (issue.getLineStart() <= 0) {
-                output.append(issue.getMessage());
-                if (StringUtils.isBlank(issue.getMessage())) {
-                    output.append(description);
+        String prismCssClass = prismLangClassFromExtension(FilenameUtils.getExtension(fileName));
+        writer.append("<pre><code class=\""+prismCssClass+" line-numbers\">");
+
+        BufferedReader reader = new BufferedReader(sourceFileReader);
+
+        List<LineRange> ranges = new ArrayList<LineRange>(); //issue.getLineRanges(); //There are only one line highlight at the time for the moment.
```

Comment:
I know that spotbugs use this concept. The primary is usually where problem will happen (NullPointerException, SQL injection, etc.) The secondary will highlight the source of the problematic variable or the duplicate condition identical to the primary line.

It would be something useful for understanding some issue but this is probably not a blocker or essential feature.